### DISCUSSION TITLE
Get: <URL>: context deadline exceeded (Client.Timeout exceeded while awaiting headers)

### DISCUSSION THREAD

**Discussion by anthonyjr2** [amidaware/tacticalrmm#1373] [General]
**Server Info (please complete the following information):**
 - OS: Ubuntu 20.04.1
 - Browser: Microsoft Edge
 - RMM Version (as shown in top left of web UI): 0.15.5

**Installation Method:**
  - [X] Standard
  - [ ] Docker

**Agent Info (please complete the following information):**
- Agent version (as shown in the 'Summary' tab of the agent from web UI): v2.4.3
- Agent OS: [e.g. Win 10 v2004, Server 2012 R2]: Windows 10 22H2

**Describe the bug**
I get this error whenever I try to install the agent on multiple different PCs outside of my LAN: 

Get: "https://xxxxxxxxxx.com/api/v3/installer/": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

It only started happening recently, I previously had no problems installing the agent. Google hasn't given me much info on this error or why it may be happening. It also doesn't work regardless of using manual installation/powershell/dynamic exe

**To Reproduce**
1) Attempt to install agent on PC external to network

**Expected behavior**
Normal agent installation

**Screenshots**
![Screenshot 2022-12-09 152735](https://user-images.githubusercontent.com/5232548/206790921-b015ad1e-e1b7-40f4-abeb-3ca5e7ccbbf2.png)

**Additional context**
I've tried restarting the RMM server and the PC hosting it, rebooting client machines, changing DNS settings, generating new .exes, nothing seems to fix it

**Comment by dinger1986** [comment]
hello!

Converted to a discussion as its working for myself and others so must be something on your setup.

Can you please provide some more information on your setup (see below)? Have you changed anything from when it stopped working? Is your cert renewed?

Pre-filling the relevant answers to these questions will also help us help you:

1. Install type? (Standard/Docker) and amount of RAM on server
  Did you deviate IN ANY WAY from these instructions? https://docs.tacticalrmm.com/install_server/
  Did you deviate IN ANY WAY from these instructions? https://docs.tacticalrmm.com/install_docker/
2. Where is the server? (VPS/onprem)
3. New/old install? Rough age of TRMM server (days/weeks/months)?

Server Install Specific questions:
4. What OS/version is TRMM installed on.
5. Using a real domain?
6. Did letsencrypt finalize and work
7. Have you looked at the troubleshooting steps? https://docs.tacticalrmm.com/troubleshooting/
8. What kind of ssl certs? Let's Encrypt, or purchased (you're not trying to make self-signed work right?)
9. Check Expiry date of your certificates in the browser (at https://rmm.example.com/ )

Network Troubleshooting
10. Are you using a proxy?
11. Are you a wizard? See https://docs.tacticalrmm.com/unsupported_guidelines/
  If so, what's in the network between agent and server?

Agent Troubleshooting
12. Is there ANY 3rd party Antivirus installed on the computer?
13. Is there any network based filtering/AV filtering?

Asking yourself questions like: When did it go from working to broken? What changed during that time? This will help you

**Reply by anthonyjr2** [reply]
Sure, here is some more info. I haven't changed anything in the last few days just suddenly I couldn't add any new agents and no matter what I tried I got this error on multiple PCs. As far as I can tell my cert is still valid because I can log into the dashboard just fine.

1. Install type? (Standard/Docker) and amount of RAM on server - Standard, 6GB
Did you deviate IN ANY WAY from these instructions? https://docs.tacticalrmm.com/install_server/ - Nope
2. Where is the server? (VPS/onprem) - On Prem
3. New/old install? Rough age of TRMM server (days/weeks/months)? - about 2 months

Server Install Specific questions:
4. What OS/version is TRMM installed on. - Ubuntu 20.04.1
5. Using a real domain? - Yes
6. Did letsencrypt finalize and work - Yes
7. Have you looked at the troubleshooting steps? https://docs.tacticalrmm.com/troubleshooting/ - Took a look but didn't see anything relating to this issue specifically.
8. What kind of ssl certs? Let's Encrypt, or purchased (you're not trying to make self-signed work right?) - Lets Encrypt
9. Check Expiry date of your certificates in the browser (at https://rmm.example.com/ ) - Not until the end of this month

Network Troubleshooting
11. Are you using a proxy? - No
12. Are you a wizard? See https://docs.tacticalrmm.com/unsupported_guidelines/ - No, this is a default install
If so, what's in the network between agent and server? There is a firewall but it hasn't been changed at all and same rules I originally created are in place. Also current agents have no issues, just new installs.

Agent Troubleshooting
13. Is there ANY 3rd party Antivirus installed on the computer? - No, just Windows Defender
14. Is there any network based filtering/AV filtering? - No

**Reply by dinger1986** [reply]
ok, can you ping api.yourdomain.com from the machine its not working on?

**Reply by anthonyjr2** [reply]
Oof, writing all this information out made me realize maybe I SHOULD check the firewall rules just in case, and yes that seems to have been the issue. Had their IPs not whitelisted. Rubber duck debugging strikes again.

Thanks for the help!

**Reply by dinger1986** [reply]
ah cool, glad its sorted
